tests/build/multi-platform.go

```diff
@@ -0,0 +1,236 @@
+			hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.platform", count)] = "linux/arm64"
+			hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.user", count)] = "ec2-user"
+			hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.secret", count)] = SecretName
+			hostConfig.Data[fmt.Sprintf("host.aws-arm64-%d.concurrency", count)] = "4"
+			count++
+		}
+
+		_, err = f.AsKubeAdmin.CommonController.CreateConfigMap(hostConfig, ControllerNamespace)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		keys := v1.Secret{}
+		keys.Name = SecretName
+		keys.Namespace = ControllerNamespace
+		keys.Labels = map[string]string{"build.appstudio.redhat.com/multi-platform-secret": "true"}
+		keys.StringData = map[string]string{"id_rsa": os.Getenv("MULTI_PLATFORM_AWS_SSH_KEY")}
+		_, err = f.AsKubeAdmin.CommonController.CreateSecret(ControllerNamespace, &keys)
+		Expect(err).ShouldNot(HaveOccurred())
+
+		trueBool := true
+		customBuildahRemotePipeline := os.Getenv(constants.CUSTOM_BUILDAH_REMOTE_PIPELINE_BUILD_BUNDLE_ENV)
+		Expect(customBuildahRemotePipeline).ShouldNot(BeEmpty())
+		ps := &buildservice.BuildPipelineSelector{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "build-pipeline-selector",
+				Namespace: testNamespace,
+			},
+			Spec: buildservice.BuildPipelineSelectorSpec{Selectors: []buildservice.PipelineSelector{
+				{
+					Name:           "custom remote-buildah selector",
+					PipelineRef:    *tekton.NewBundleResolverPipelineRef("buildah-remote-pipeline", customBuildahRemotePipeline),
+					WhenConditions: buildservice.WhenCondition{DockerfileRequired: &trueBool},
+				},
+			}},
+		}
+		Expect(f.AsKubeAdmin.CommonController.KubeRest().Create(context.TODO(), ps)).To(Succeed())
+
+		timeout = time.Minute * 20
+		interval = time.Second * 10
+
+		applicationName = fmt.Sprintf("multi-platform-suite-application-%s", util.GenerateRandomString(4))
+		app, err := f.AsKubeAdmin.HasController.CreateApplication(applicationName, testNamespace)
+		Expect(err).NotTo(HaveOccurred())
+		Expect(utils.WaitUntil(f.AsKubeAdmin.HasController.ApplicationGitopsRepoExists(app.Status.Devfile), 30*time.Second)).To(
+			Succeed(), fmt.Sprintf("timed out waiting for gitops content to be created for app %s in namespace %s: %+v", app.Name, app.Namespace, err),
+		)
+
+		componentName = fmt.Sprintf("multi-platform-suite-component-%s", util.GenerateRandomString(4))
+
+		// Create a component with Git Source URL being defined
+		componentObj := appservice.ComponentSpec{
+			ComponentName: componentName,
+			Source: appservice.ComponentSource{
+				ComponentSourceUnion: appservice.ComponentSourceUnion{
+					GitSource: &appservice.GitSource{
+						URL:      multiPlatformProjectGitUrl,
+						Revision: multiPlatformProjectRevision,
+					},
+				},
+			},
+		}
+		component, err = f.AsKubeAdmin.HasController.CreateComponent(componentObj, testNamespace, "", "", applicationName, true, map[string]string{})
+		Expect(err).ShouldNot(HaveOccurred())
+	})
```

Comment:
Does it make sense to check remote instances for readiness before starting the suite?

Reply:
I don't think it matters, it will fail either way, and we don't want to skip if there are no instances as it would mean that if the instances fell over it would not be obvious that the tests were not being run.